### ISSUE TITLE
PET示例中，模型转换为静态图后输出维度与动态图不一致

### ISSUE THREAD

**Issue by lastrei** [PaddlePaddle/PaddleNLP#2656]
欢迎您反馈PaddleNLP使用问题，非常感谢您对PaddleNLP的贡献！
在留下您的问题时，辛苦您同步提供如下信息：
- 版本、环境信息
1）PaddleNLP和PaddlePaddle版本：
paddlepaddle-gpu               2.2.2.post112
paddlenlp                      2.3.2
2）系统环境：请您描述系统类型
windows 11
python 3.9
PET算法示例中，在测试输入'啊啊啊啊'后
prediction_scores输出维度为[2, 18000]，这个是正确的[lable_length,vocab_size]
https://github.com/PaddlePaddle/PaddleNLP/blob/7e1a4332d543ae40becdc0436b6deab614127fc7/examples/few_shot/pet/predict.py#L137

将模型转换为静态图代码如下:

from model import ErnieForPretraining, ErnieMLMCriterion
import paddle
import os

model = ErnieForPretraining.from_pretrained('ernie-1.0')
state_dict = paddle.load('./checkpoint/model_160/model_state.pdparams')
model.set_dict(state_dict)
print("Loaded parameters over ")
model.eval()

model = paddle.jit.to_static(
    model,
    input_spec=[
        paddle.static.InputSpec(
            shape=[None, None], dtype="int64", name='src_ids'),  # input_ids
        paddle.static.InputSpec(
            shape=[None, None], dtype="int64", name='token_type_ids'),  # segment_ids
        paddle.static.InputSpec(
            shape=[None, None], dtype="int64", name='masked_positions')  # masked_positions
    ])

save_path = os.path.join('./infer_model/', "inference")
paddle.jit.save(model, save_path)

转换完毕后 进行预测，静态模型的output_handle输出维度为[1,15,18000] 即[batch_size,sentence_length,vocab_size]
，维度不对后就无法进行下面的将prob转换为label的计算，infer_predict的代码请参见附件，麻烦各位了~

[infer_predict.txt](https://github.com/PaddlePaddle/PaddleNLP/files/8990941/infer_predict.txt)

**Comment by wawltor** [comment]
你取出来的是第几个结果了，这个模型的输出是有两个输出结果的

**Comment by lastrei** [comment]
> 你取出来的是第几个结果了，这个模型的输出是有两个输出结果的

检查了下静态模型,print(self.predictor.get_output_names())
只有一个返回['tmp_42']呀。。

在模型组网的时候，forward下有两个输出prediction_scores, seq_relationship_score，但是最终返回的只有prediction_scores
转换成静态图之后也就只有1个了

**Comment by wawltor** [comment]
> > 你取出来的是第几个结果了，这个模型的输出是有两个输出结果的
> 
> 检查了下静态模型,print(self.predictor.get_output_names()) 只有一个返回['tmp_42']呀。。
> 
> 在模型组网的时候，forward下有两个输出prediction_scores, seq_relationship_score，但是最终返回的只有prediction_scores 转换成静态图之后也就只有1个了

看了一下模型只有一个输出了，你可以在模型上zuo做一下相应的改动
<img width="717" alt="image" src="https://user-images.githubusercontent.com/16698950/176079709-9b32c14e-bf07-48a8-b7ff-2a022f163eb6.png">

**Comment by lastrei** [comment]
> > > 你取出来的是第几个结果了，这个模型的输出是有两个输出结果的
> > 
> > 
> > 检查了下静态模型,print(self.predictor.get_output_names()) 只有一个返回['tmp_42']呀。。
> > 在模型组网的时候，forward下有两个输出prediction_scores, seq_relationship_score，但是最终返回的只有prediction_scores 转换成静态图之后也就只有1个了
> 
> 看了一下模型只有一个输出了，你可以在模型上zuo做一下相应的改动 <img alt="image" width="717" src="https://user-images.githubusercontent.com/16698950/176079709-9b32c14e-bf07-48a8-b7ff-2a022f163eb6.png">

您的意思是把forward的输出改为输出prediction_sc和seq_relationship_sc么？..

**Comment by wawltor** [comment]
> > > > 你取出来的是第几个结果了，这个模型的输出是有两个输出结果的
> > > 
> > > 
> > > 检查了下静态模型,print(self.predictor.get_output_names()) 只有一个返回['tmp_42']呀。。
> > > 在模型组网的时候，forward下有两个输出prediction_scores, seq_relationship_score，但是最终返回的只有prediction_scores 转换成静态图之后也就只有1个了
> > 
> > 
> > 看了一下模型只有一个输出了，你可以在模型上zuo做一下相应的改动 <img alt="image" width="717" src="https://user-images.githubusercontent.com/16698950/176079709-9b32c14e-bf07-48a8-b7ff-2a022f163eb6.png">
> 
> 您的意思是把forward的输出改为输出prediction_sc和seq_relationship_sc么？..

嗯嗯，你可以增加一个输出或者把原本的输出改成你想要的输出也行

**Comment by lastrei** [comment]
还是有问题 ，这里我加了一个paddle.nn.Softmax() forward输出softmax_fn(prediction_scores)
转换后报维度不对X = [1, 35, 768] and the shape of Y = [2, 768]. Received [35] in X is not equal to [2]
这里的35还是输入的句长而不是label_length

**Comment by lastrei** [comment]
已经解决了，问题出在定义forward入参的时候，定义了position_ids和mask_attention=None,虽然没有使用到，但是在export到静态图的时候，没有添加这两个入参，造成了在infer的时候position_ids代替了masked_positions，造成了维度不对的问题

所以问问大佬，我导出的时候定义入参paddle.static.InputSpec(
            shape=[None], dtype="int64", name='masked_positions')这样使用了name的

在infer的时候我使用predictor.get_input_names()获取对应的入参名称进行赋值，为什么还是会出现把masked_positions当成了position_ids传值....

**Comment by lastrei** [comment]
<img width="463" alt="767b39e810469a303000c81fd4cbef1" src="https://user-images.githubusercontent.com/32331131/176090124-70683995-9980-45e2-8afb-5c798b84f53e.png">
已解决，感谢
